viewer/packages/camera-manager/src/Flexible/FlexibleCameraManager.ts

```diff
@@ -0,0 +1,552 @@
+    // If an object is picked, zoom in to the object (the target will be in the middle of the bounding box)
+    if (raycastResult.pickedBoundingBox !== undefined) {
+      const { position, target } = fitCameraToBoundingBox(this.camera, raycastResult.pickedBoundingBox, 3);
+      moveCameraTo(this, position, target, this.options.animationDuration);
+      return;
+    }
+    // If not particular object is picked, set camera position half way to the target
+    const newTarget =
+      raycastResult.intersection?.point ??
+      this.getTargetByBoundingBox(event.offsetX, event.offsetY, raycastResult.modelsBoundingBox);
+
+    const newPosition = new Vector3().subVectors(newTarget, this.camera.position);
+    newPosition.divideScalar(2);
+    newPosition.add(this.camera.position);
+    moveCameraTo(this, newPosition, newTarget, this.options.animationDuration);
+  }
+
+  //================================================
+  // INSTANCE METHODS: Updates
+  //================================================
+
+  private updateCameraNearAndFar(): void {
+    // See https://stackoverflow.com/questions/8101119/how-do-i-methodically-choose-the-near-clip-plane-distance-for-a-perspective-proj
+    if (this._isDisposed) {
+      return;
+    }
+    if (!this.options.automaticNearFarPlane) {
+      return;
+    }
+    CameraManagerHelper.updateCameraNearAndFar(this.camera, this._currentBoundingBox);
+  }
+
+  private updateControlsSensitivity(): void {
+    if (this._isDisposed) {
+      return;
+    }
+    if (!this.options.automaticSensitivity) {
+      return;
+    }
+    // This is used to determine the speed of the camera when flying with ASDW.
+    // We want to either let it be controlled by the near plane if we are far away,
+    // but no more than a fraction of the bounding box of the system if inside
+    const diagonal = this.getBoundingBoxDiagonalXY();
+    const diagonalFraction = diagonal * this.options.sensitivityDiagonalFraction;
+    const nearFraction = 0.1 * this.camera.near;
+
+    let sensitivity: number;
+    if (diagonalFraction < nearFraction) {
+      sensitivity = nearFraction;
+    } else {
+      sensitivity = this.options.getLegalSensitivity(diagonalFraction);
+    }
+    this.options.sensitivity = sensitivity;
+
+    // Nils: Need this for a while to fine tune
+    // console.log('diagonalFraction:     ', diagonalFraction);
+    // console.log('nearFraction:         ', nearFraction);
+    // console.log('sensitivity:          ', sensitivity);
+  }
+
+  public getBoundingBoxDiagonal(): number {
+    return getDiagonal(this._currentBoundingBox);
+  }
+  public getBoundingBoxDiagonalXY(): number {
+    return getDiagonalXY(this._currentBoundingBox);
+  }
+}
+
+function getDiagonal(boundingBox: Box3): number {
+  return boundingBox.min.distanceTo(boundingBox.max);
+}
+function getDiagonalXY(boundingBox: Box3): number {
+  return Math.sqrt((boundingBox.max.x - boundingBox.min.x) ** 2 + (boundingBox.max.y - boundingBox.min.y) ** 2);
```

Comment:
Now that I think of it - is this supposed to be the horizontal plane of the bounding box? Y is the up-direction in ThreeJS-space (and therefore, Reveal-space)

Reply:
Fixed